client/src/views/users/utils.ts

```diff
@@ -0,0 +1,105 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { SmallDisplayCategoryUserContextMenu, SmallDisplayUserContextMenu } from '@/components/small-display';
+import { UserModel } from '@/components/users';
+import { UserProfile } from '@/parsec';
+import UserContextMenu, { UserAction } from '@/views/users/UserContextMenu.vue';
+import { modalController, popoverController } from '@ionic/vue';
+
+async function canUpdateProfile(user: UserModel, users: UserModel[], clientIsAdmin: boolean): Promise<boolean> {
+  if (!clientIsAdmin) {
+    return false;
+  }
+  if (users.length === 0) {
+    return user.currentProfile !== UserProfile.Outsider && !user.isRevoked();
+  } else if (users.length === 1) return users[0].currentProfile !== UserProfile.Outsider && !users[0].isRevoked();
+  else {
+    return users.some((u) => u.currentProfile !== UserProfile.Outsider && !u.isRevoked());
+  }
+}
+
+async function canRevoke(user: UserModel, users: UserModel[], clientIsAdmin: boolean): Promise<boolean> {
+  if (!clientIsAdmin) {
+    return false;
+  }
+  if (users.length === 0) {
+    return user.isRevoked() === false;
+  } else if (users.length === 1) {
+    return users[0].isRevoked() === false;
+  } else {
+    return users.some((u) => !u.isRevoked());
+  }
+}
+
+export async function openUserContextMenu(
+  event: Event,
+  user: UserModel,
```

Comment:
Why have `user` and `selectedUsers` separated? Can't we just have `users: Array<UserModel>`?